Fixes TV > Tpl Access sorting bug

davidpede opened the pull request:
### What does it do?
Fixes v old bug by setting remoteSort to false in the 'TV>Template Access' grid

### Why is it needed?
When sorting by the Access column the processor returns no results and xpdo logs an unknown column error. This because the access value is set manually in the processor after the db query as it is not stored in db table: https://github.com/modxcms/revolution/blob/5c59d5bc4f2ffd77454d2a42a1d1046ba2f7d808/core/model/modx/processors/element/tv/template/getlist.class.php#L101

### Related issue(s)/PR(s)
Fixes: #12319, #7896

opengeek committed:
Change new installs to create tables with InnoDB engine

opengeek committed:
Attempt to use the beta Trusty Travis CI environment

opengeek committed:
Add missing return statements to built-in validation classes

opengeek committed:
Make InnoDB the default engine for MySQL

opengeek committed:
Remove PHP 5.3 from Travis CI version matrix

opengeek committed:
Generate engine into the map files for MySQL

opengeek committed:
Regenerate map files for MySQL with engine explicitly declared

opengeek committed:
Conditionally override engine to MyISAM if MySQL < 5.6

opengeek committed:
Force engine override to MyISAM if MySQL server version cannot be detected

opengeek committed:
Reduce varchar and text index prefixes for utf8mb4 support in mysql

davidpede committed:
Enable sorting by 'assigned' column in template variable grid

Jako commented:
Instead of disabling remote sort, the access column should not be sortable in https://github.com/davidpede/revolution/blob/ed8038bfdc75d840d2d624a430a665b725748b88/manager/assets/modext/widgets/element/modx.grid.tv.template.js#L15

Jako commented:
But before you disable sortable of the access column: How expensive would it be to create a sortable `template_access` column in the query and reference it in the grid?

davidpede commented:
@Jako How does last commit look for a quick solution? It seem to work as `tmplvarid` is either the current tv id or null for that query

Jako commented on core/model/modx/processors/element/tv/template/getlist.class.php:
```diff
@@ -27,6 +27,11 @@ public function initialize() {
             'tv' => false,
         ));
+        if ($this->getProperty('sort') == 'access') {
+          $this->setProperties(array(
```
$this->setProperty('sort', 'tmplvarid');

OptimusCrime committed:
Return better error message if group name already exists

Jako reviewed (commented):
Should be almost suitable. But since `tmplvarid` could be an integer, the sorting depends not only on true/false but on the integer value.

You could add a select query with 
```
$c->select(array('access' => 'IF(`TemplateVarTemplates.tmplvarid`,true,false)'));
```
in the getData method, but I don't know wether this is only MySQL syntax.

opengeek committed:
Handle version detection appropriately for MariaDB

opengeek committed:
Modify uri index prefix length to 255 from 333 for InnoDB support

Mark-H committed:
Add support for new_folder_permissions_cache and new_file_permissions_cache settings to control cache permissions. Make sure scripts cache uses right file permissions. #12677

Merge remote-tracking branch 'upstream/pr/12677' into 2.x

* upstream/pr/12677:
  revert changes in xpdo file
  don't reset $options array
  make sure that the includes folder for scripts also uses new cache permission settings
  better names for new settings
  fix errors in travis-ci ($mode needs to be set)
  introduce 2 new settings for file/folder permissions of file-cache
  use new_file_permission setting in writeFile
  always force the disired folder permissions

Mark-H committed:
Add new modDirectory->getFiles() method to list files/folders in a directory [#13096]

Merge remote-tracking branch 'upstream/pr/13096' into 2.x

* upstream/pr/13096:
  Fix Call to undefined method SplFileInfo::isDot() error when using recursive scanning
  Implement feedback: correct docblock, decrease if nesting, use call_user_func instead of clojure only, skip empty callback return values
  Rename method again, list is illegal...no getList()
  Improve sort, enable multibyte, allow listing of folders, micro optimizations, rename method to list()
  Shorten/optimize if/else logic
  Add ability to use a callback function to modify the output on the fly
  Add ability to skip individual files
  Make skiphidden option more reliable
  Rebase on 2.x and refactor method
  Fix docblock description
  Improving the method based on feedback and make it compatible with PHP <= 5.3
  modFileHandler enhancement: added modDirectory getFiles() method

Mark-H committed:
Some modRest refactoring to clean up code styles and fix doctypes [#13133]

Merge remote-tracking branch 'upstream/pr/13133' into 2.x

* upstream/pr/13133:
  Right phpdoc type for $parameters
  Allow to switch off the checkForSuffix test on base of a service setting.
  Fixing Method not allowed for not numeric IDs $id could be a string that does not start with a number. In that case the check with integer casting is not valid. It should be enough if $id is not empty.
  Add status code and change the order of the response array to get the same answer as when the API endpoint not exists
  Wrong phpDoc type in recursive method call

Mark-H committed:
Fix output filter handling of non-existent TV tags to be consistent with how placeholders are handled [#13203]

Merge remote-tracking branch 'upstream/pr/13205' into 2.x

* upstream/pr/13205:
  Add test for new behavior
  Ignore composer related file and dir
  Handle non-existing TVs as if they are empty, instead of ignoring them

Mark-H committed:
Automatically change to the resource tab that holds an error when encountering a validation error saving a resource [#13202]

Merge remote-tracking branch 'upstream/pr/13206' into 2.x

* upstream/pr/13206:
  Fix mistakes, implement handling for duplicate alises
  Make automatic tab switcher handle any kind of invalid state
  Automatically change tab to settings if failure and somewhere else

Mark-H committed:
Move OnFileManagerBeforeUpload event so it can also be used to prevent uploads or change file info [#13067]

Merge remote-tracking branch 'upstream/pr/13067' into 2.x

* upstream/pr/13067:
  move OnFileManagerBeforeUpload event call

Mark-H committed:
Lower memory usage of duplicating contexts with lots of children [#13217]

Merge remote-tracking branch 'upstream/pr/13285' into 2.x

* upstream/pr/13285:
  Update modresource.class.php
  Update duplicate.class.php

opengeek committed:
Change new installs to create tables with InnoDB engine on mysql

Merge remote-tracking branch 'origin/pr/13462' into 2.x

* origin/pr/13462:
  Modify uri index prefix length to 255 from 333 for InnoDB support
  Handle version detection appropriately for MariaDB
  Force engine override to MyISAM if MySQL server version cannot be detected
  Conditionally override engine to MyISAM if MySQL < 5.6
  Regenerate map files for MySQL with engine explicitly declared
  Generate engine into the map files for MySQL
  Remove PHP 5.3 from Travis CI version matrix
  Make InnoDB the default engine for MySQL
  Add missing return statements to built-in validation classes
  Attempt to use the beta Trusty Travis CI environment
  Change new installs to create tables with InnoDB engine

opengeek committed:
Reduce varchar and text index prefixes for utf8mb4 support in mysql

Merge remote-tracking branch 'origin/pr/13559' into 2.x

* origin/pr/13559:
  Reduce varchar and text index prefixes for utf8mb4 support in mysql

Jako committed:
cleanAlias on base of context settings

Jako committed:
Check $this->xpdo for instance of modX

Commit:
Fixed code style.

Mark-H committed:
Make sure an invalid ctx parameter (array instead of string) doesn't throw errors

opengeek committed:
Make ContextResource optional in query for rebuilding contexts

Merge remote-tracking branch 'origin/pr/13360' into 2.x

* origin/pr/13360:
  Fixed code style.
  Fixed warning output.
  Fixed missing context condition when query is optimized.
  Added function to turn off inefficient queries in modcontext. Added system setting to disable queries, added lexicon entries for setting.

opengeek committed:
Make searchbar accessible via assistive tech landmarks

Merge remote-tracking branch 'origin/pr/13437' into 2.x

* origin/pr/13437:
  Added some a11y recommendation as per #13436

opengeek committed:
Replace all hardcoded http versions by $_SERVER['SERVER_PROTOCOL']

Merge remote-tracking branch 'origin/pr/13518' into 2.x

* origin/pr/13518:
  Replace all hardcoded http versions by $_SERVER['SERVER_PROTOCOL'] variable

opengeek committed:
Improve description of TemplateVariable Input Option Values

Merge remote-tracking branch 'origin/pr/13550' into 2.x

* origin/pr/13550:
  Fixed missed `=`
  Little improving of field captions on TV creating page

opengeek committed:
Search improvements for user management

Merge remote-tracking branch 'origin/pr/13551' into 2.x

* origin/pr/13551:
  Improve search in user management

opengeek committed:
Add search/filtering to plugin event list

Merge remote-tracking branch 'origin/pr/13552' into 2.x

* origin/pr/13552:
  Removed console.log call
  Added filtering and search to event list during creting a new plugin

opengeek committed:
Removing null-byte character check

Merge remote-tracking branch 'origin/pr/13581' into 2.x

* origin/pr/13581:
  Removing null-byte character check #13289

bezumkin committed:
Fix for #13631

opengeek committed:
Fix site_status issue when a session is not available

opengeek committed:
Add stream upload support for binary files to modRestService

Merge remote-tracking branch 'origin/pr/13164' into 2.x

* origin/pr/13164:
  Move the $params check
  Reformatting code
  Typo
  Added stream based uploads and file handle for binary uploads - the file upload is changed to support streams - the handle for the stream is set as parameter value for the modRestController and could be handled there

opengeek committed:
Enable rendering of SVG previews in Media Browser

opengeek committed:
Additional SVG preview improvements

- Add SVG support for IE and Edge
- Add proportional SVG thumbnail scaling
- Add SVG support to phpthumb for S3
- Check background for transparent images

opengeek committed:
Add new Who's Online dashboard widget

Squashed commit of the following:

commit 854eb97ecb3331dce23014e84c0ba3a99e1eb295
Author: Jason Coward <jason@opengeek.com>
Date:   Thu Oct 12 13:53:47 2017 -0600

    Add changelog entry

commit cf7fa3c00e01a788d07bea90939b5e13de7ae516
Author: David Pede <david@tasian.media>
Date:   Mon Jul 10 22:40:27 2017 +0100

    New getonline processor

    Update processor

    Update tpl

    Update controller

    New grid

    Add 'manager_date_format' and 'manager_time_format' setting support

    Fix hardcoded datetime zone

    Remove date from current time

    Add 'manager_use_fullname' setting support

    Lexicon load in addHtml

    Add pagination to grid

Jako committed:
Add OPTIONS request method
Used in cross-origin HTTP request
- https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS
- http://zacstewart.com/2012/04/14/http-options-method.html

An additional class property $allowedMethod is introduced to be filled with an array of allowed request methods.
Another additional class properties $allowedMethod could be filled with an array of allowed request headers.

I am not sure about allowing the OPTIONS request method in authentificated controllers, so this part in the modRestService class could be removed.

opengeek committed:
Add OnResourceCacheUpdate event

Merge remote-tracking branch 'origin/pr/13590' into 2.x

* origin/pr/13590:
  Invoke event when resource cache is updated + add event to build file

opengeek committed:
Add CSS class to TV containers

Merge remote-tracking branch 'origin/pr/13602' into 2.x

* origin/pr/13602:
  Add class to TV containers

opengeek committed:
Fetch Lexicon languages and topics from DB as well as file system

opengeek committed:
Fix endless loop when error log is too big

Merge remote-tracking branch 'origin/pr/13632' into 2.x

* origin/pr/13632:
  Fix for #13631

opengeek committed:
Fix site_status issue when a session is not available

Merge remote-tracking branch 'origin/pr/13635' into 2.x

* origin/pr/13635:
  Fix site_status issue when a session is not available

opengeek committed:
Make buttongroup visible again
- Fixing #13612 caused by #13543

opengeek committed:
Fix processing of noncacheable elements inside cached

Merge remote-tracking branch 'origin/pr/13530' into 2.x

* origin/pr/13530:
  Proper processing uncached elements inside cached

achterbahn committed:
Fixes issue #13639

achterbahn committed:
Fixing issue 13640. Was already discussed here: https://github.com/modxcms/revolution/pull/13245/files/c43a25483b0e39a0a6e8fa700f7f96ecc33475d1#r95083093

opengeek committed:
Prevent connector errors from invalid ctx parameter

Merge remote-tracking branch 'origin/pr/13627' into 2.x

* origin/pr/13627:
  Make sure an invalid ctx parameter (array instead of string) doesn't throw errors

opengeek committed:
Fix format of chunk title

Merge remote-tracking branch 'origin/pr/13643' into 2.x

* origin/pr/13643:
  Fixes issue #13639

opengeek committed:
Fix redirect when deleting elements

Merge remote-tracking branch 'origin/pr/13644' into 2.x

* origin/pr/13644:
  Fixing issue 13640. Was already discussed here: https://github.com/modxcms/revolution/pull/13245/files/c43a25483b0e39a0a6e8fa700f7f96ecc33475d1#r95083093

opengeek committed:
Add OPTIONS request method to modRestController

Merge remote-tracking branch 'origin/pr/13636' into 2.x

* origin/pr/13636:
  Add OPTIONS request method Used in cross-origin HTTP request - https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS - http://zacstewart.com/2012/04/14/http-options-method.html

Jako committed:
Merge all context settings and pass them as options to filterPathSegment

opengeek committed:
Enable context setting overrides in modResource->cleanAlias()

Merge remote-tracking branch 'origin/pr/13622' into 2.x

* origin/pr/13622:
  Check $this->xpdo for instance of modX
  cleanAlias on base of context settings

opengeek committed:
Forgot to pull first — sigh
Merge remote-tracking branch 'origin/pr/13622' into 2.x

* origin/pr/13622:
  Merge all context settings and pass them as options to filterPathSegment

SintezCode committed:
Hide empty template variable tabs in the resource panel

Jako committed:
Adding padding top for new .modx-alert and .modx-confirm classes

Jako committed:
Purge old packages button in packages management

Jako committed:
Refactor setUserGroups/addUsers methods
- Don't remove the user from all existing groups and add it afterwards to all belonging groups
- Invoke 'OnUserBeforeAddToGroup' and 'OnUserAddToGroup'

Jako committed:
Wrong indent

Jako committed:
Add .less, .scss, .sass and .css.map as default allowed upload file types [#13592]

Merge remote-tracking branch 'upstream/pr/13592' into 2.x

* upstream/pr/13592:
  Add .less, .scss and .css.map as default allowed upload file types

Mark-H committed:
Sync 2.5 and 2.x changelogs to ensure a complete history

Jako committed:
Sync 2.5 and 2.x changelogs to ensure a complete history

* upstream/pr/13655:
  Sync 2.5 and 2.x changelogs to ensure a complete history

opengeek committed:
Update translations

opengeek committed:
Revert "Sync 2.5 and 2.x changelogs to ensure a complete history"

This reverts commit c0ae9f94e4469248ebba86fa8e67c8441c32d424.

opengeek committed:
Improve setUserGroups/addUsers methods

Merge remote-tracking branch 'origin/pr/13653' into 2.x

* origin/pr/13653:
  Refactor setUserGroups/addUsers methods - Don't remove the user from all existing groups and add it afterwards to all belonging groups - Invoke 'OnUserBeforeAddToGroup' and 'OnUserAddToGroup'

opengeek committed:
Add top padding to .modx-alert and .modx-confirm classes

Merge remote-tracking branch 'origin/pr/13652' into 2.x

* origin/pr/13652:
  Wrong indent
  Adding padding top for new .modx-alert and .modx-confirm classes

Commit:
Add alias_visible checkbox to quick update resource form

Commit:
Fix typo

Commit:
Merge branch '2.x' into feature/exclude-resource-alias-for-child

opengeek committed:
Update translations

opengeek committed:
Update welcome page URL

Commit:
Fix alias_visible checkbox in quick create/update window

Commit:
Update translation (taken from evo)

rtripault committed:
Fixed issue introduced by #13506 preventing to download 2 packages in a row

opengeek committed:
Update version for 2.6.0-pl release

opengeek committed:
Update version for 2.6.1 development

Jako committed:
Fix javascript issue on resources containing a quote sign
- Introduced in #13497

Commit:
Reset image_width and image_height, remove introduced file_width and file_height

Jako committed:
Missed documentation lines

Jako committed:
Purge old versions of one package

achterbahn committed:
Issue #13668: Fixed wrong title format in different Manager pages

achterbahn committed:
Merge remote-tracking branch 'upstream/2.x' into 2.x

Jako committed:
Replace newline with br tag

Jako committed:
Revert newline patch and add white-space style

and some other small beautification

opengeek committed:
Fix invokeEvent call for new OnResourceCacheUpdate event

Squashed commit of the following:

commit 1d0f54f1cde7de12d7e27fb70180f80d52a16e96
Author: Jason Coward <jason@opengeek.com>
Date:   Tue Nov 7 13:19:25 2017 -0700

    Add changelog entry

commit b8610eeb0432435fafbcc196e16e8678b87bc29c
Author: Володя <vgrish@gmail.com>
Date:   Mon Nov 6 10:33:14 2017 +0300

    Update modresource.class.php

    - fix error

opengeek committed:
Fix CLI installation on MySQL < 5.6 to use MyISAM engine

theboxer committed:
Check if contentEl exists before using it

matdave committed:
Allow callback if nothing is selected

Would not fire browser callback if cancel hit or nothing was selected. Fixes TinyMCE RTE Issue 10 https://github.com/modxcms/tinymce-rte/issues/10

opengeek committed:
Fix console error when editing resources with tv tab

Merge remote-tracking branch 'origin/pr/13683' into 2.6.x

* origin/pr/13683:
  Check if contentEl exists before using it

opengeek committed:
Fix javascript issue on resources containing quotes

Merge remote-tracking branch 'origin/pr/13669' into 2.6.x

* origin/pr/13669:
  Fix javascript issue on resources containing a quote sign - Introduced in #13497

opengeek committed:
Fix title format in various manager views

Merge remote-tracking branch 'origin/pr/13675' into 2.6.x

* origin/pr/13675:
  Issue #13668: Fixed wrong title format in different Manager pages

opengeek committed:
Fix CLI installation to properly detect MySQL server version

Merge remote-tracking branch 'origin/pr/13680' into 2.6.x

* origin/pr/13680:
  Fix CLI installation on MySQL < 5.6 to use MyISAM engine

Jako committed:
Change the style to pre-wrap

Jako committed:
Revert behavior of image_width and image_height for media source images

* upstream/pr/13672:
  Reset image_width and image_height, remove introduced file_width and file_height

opengeek committed:
Improve changelog display in package browser

Merge remote-tracking branch 'origin/pr/13677' into 2.6.x

* origin/pr/13677:
  Change the style to pre-wrap
  Revert newline patch and add white-space style
  Replace newline with br tag

Jako committed:
Fixing 'Flush Your Permissions' in topmenu

opengeek committed:
Fix Flush Your Permissions top menu item

Merge remote-tracking branch 'origin/pr/13690' into 2.6.x

* origin/pr/13690:
  Fixing 'Flush Your Permissions' in topmenu

Jako committed:
Allow callback if nothing is selected in MODx.browser

Merge remote-tracking branch 'upstream/pr/13684' into 2.6.x

* upstream/pr/13684:
  Allow callback if nothing is selected

Commit:
add output modifier filterPathSegment

Commit:
change modifier name

Commit:
Adding system setting "inherit_parent_template" to fix #5980

Commit:
Revert "change modifier name"

This reverts commit 7bd1a2a89b9706418e945809aae059dc7bf51f2a.

Commit:
only thumbnail files that can be thumbnailed

Commit:
add error messages logging

Commit:
change type of $preview from int to boolean for files tree

Commit:
change check of $preview to integer for media browser grid

OptimusCrime committed:
Make sure Widget is instance of modDashboardWidget before calling methods

OptimusCrime committed:
Show correct icon in search results for non-document resource types

opengeek committed:
Avoid fatal error if a modFieldTag is processed without a resource

MrRoco committed:
Symlink redirect loop

Prevent an infinite loop when symlink equals the id of the resource.

opengeek committed:
Show correct Resource type icon in search results

Merge remote-tracking branch 'origin/pr/13705' into 2.6.x

* origin/pr/13705:
  Show correct icon in search results for non-document resource types

opengeek committed:
Fix phpthumb issue in files tree and media browser

Merge remote-tracking branch 'origin/pr/13704' into 2.6.x

* origin/pr/13704:
  change check of $preview to integer for media browser grid
  change type of $preview from int to boolean for files tree
  add error messages logging
  only thumbnail files that can be thumbnailed

opengeek committed:
Ensure dashboard widget exists before calling methods on it

Merge remote-tracking branch 'origin/pr/13604' into 2.6.x

* origin/pr/13604:
  Make sure Widget is instance of modDashboardWidget before calling methods

achterbahn committed:
Merge remote-tracking branch 'upstream/2.x' into 2.x

Fi1osof committed:
Update modx.class.php

Allow work with 3rd-party tables without prefixes.

Jako committed:
Get only unique template paths

opengeek committed:
Get only unique template paths for manager controllers

Merge remote-tracking branch 'origin/pr/13717' into 2.6.x

* origin/pr/13717:
  Get only unique template paths

opengeek committed:
Update lexicon translations

opengeek committed:
Prevent infinite loop when a modSymLink refers to itself

Merge remote-tracking branch 'origin/pr/13710' into 2.6.x

* origin/pr/13710:
  Symlink redirect loop

opengeek committed:
Remove concatenated key from name field in Contexts grid

Merge remote-tracking branch 'origin/pr/13372' into 2.6.x

* origin/pr/13372:
  Edit Fixed #13371
  Fixed #13371

achterbahn committed:
Bugfix 13702: FileManager/MediaBrowser don't show files. Whitespace causing problems in setting "allowedFileTypes" in MediaSource

opengeek committed:
Increase efficiency of cache refresh on autopublish events

Merge remote-tracking branch 'origin/pr/13572' into 2.6.x

* origin/pr/13572:
  Fix too aggressive cache clearing when triggering an autopublish event

opengeek committed:
Update version for 2.6.1-pl release

matdave committed:
Adding manager log to the resource changes tab

`classKey: 'mod%o%u'` is due to issue  #13733

matdave committed:
Tabbing error on line 112

matdave committed:
Add Manager Log Grid to Resource Data

Adding a reference to the manager log grid source file for inclusion on the changes tab.

Commit:
Modify getList processor to allow comma separated classKeys, and updating Changes tab action grid to request based on assigned class key as well as modResource

matdave committed:
Changed from boolean to turn off parent inheritance to work more like Evo with the additional option of inheriting sibling templates

Mark-H committed:
Support E_LEVEL constants in debug system setting and calls to $modx->setDebug. Fix "1" not being recognised as boolean true.

Merge remote-tracking branch 'upstream/pr/12590' into 2.x

* upstream/pr/12590:
  Add support for CONSTANT format for "debug" settings (#12579)

Mark-H committed:
Use a stricter type check in the resource tree to avoid potential uncaught errors in the resource tree #13262

Merge remote-tracking branch 'upstream/pr/13262' into 2.x

* upstream/pr/13262:
  Fix issue - not showing resources tree in modx manager.

Mark-H committed:
Allow plugins OnDocFormRender to set the template to use with $resource->set('template', 3) like other fields

Merge remote-tracking branch 'upstream/pr/13330' into 2.x

* upstream/pr/13330:
  Fixed bug #13049 #modxbughunt

Mark-H commented:
Sounds this is still pending improvements, @davidpede?

Mark-H committed:
Add "Purge Old Versions" to a packages' version listing, to clean up old versions #12818 #13654

Merge remote-tracking branch 'upstream/pr/13654' into 2.x

* upstream/pr/13654:
  Purge old versions of one package
  Missed documentation lines
  Purge old packages button in packages management

Mark-H committed:
New resource option "Use current alias in alias path" to allow hiding resources from the URI [#11153, #13660]

Merge remote-tracking branch 'upstream/pr/13660' into 2.x

* upstream/pr/13660:
  Update translation (taken from evo)
  Fix alias_visible checkbox in quick create/update window
  Fix typo
  Add alias_visible checkbox to quick update resource form
  Refresh children URIs on save/update when alias_visible is marked as dirty
  Implement logic for hiding parent alias fragments
  Restate resource modext changes
  Undo accidental changes to modResource
  Set default checkbox value to checked
  Add english lexicon entries
  Move checkbox to the left side and add help text + label
  Add alias_visible checkbox to resource form
  Add upgrade script for new column field
  Add property to modResource class
  Add alias_visible to map files

Mark-H committed:
Add "filterPathSegment" output filter to turn a string into url-safe string [#13699]

Merge remote-tracking branch 'upstream/pr/13699' into 2.x

* upstream/pr/13699:
  Revert "change modifier name"
  change modifier name
  add output modifier filterPathSegment

Mark-H committed:
Make sure requests to containers without the container suffix are redirected to the right url with container suffix [#13142]

Merge remote-tracking branch 'upstream/pr/13142' into 2.x

* upstream/pr/13142:
  Fix $containerSuffix is appended twice

Mark-H committed:
Ignore spaces in allowedExtensions properties and relevant system settings to ensure the right file types show up [#13702]

Merge remote-tracking branch 'upstream/pr/13721' into 2.x

* upstream/pr/13721:
  Bugfix 13702: FileManager/MediaBrowser don't show files. Whitespace causing problems in setting "allowedFileTypes" in MediaSource

Mark-H committed:
Add list of recent manager log entries to the Resource Overview page [#13734]

Merge remote-tracking branch 'upstream/pr/13734' into 2.x

* upstream/pr/13734:
  Modify getList processor to allow comma separated classKeys, and updating Changes tab action grid to request based on assigned class key as well as modResource
  Add Manager Log Grid to Resource Data
  Tabbing error on line 112
  Adding manager log to the resource changes tab

davidpede commented:
@Mark-H Yes leave this for now, I'll look at jako's suggestions asap

sdrenth committed:
Fixing complete step during setup would always be in EN language

Finetuned committed:
Addressed  issue #13309 #modxbughunt
Removed the maxSize restriction on the west region allowing the splitbar to be moved further than 800px

Finetuned committed:
removed the  maxSize property from the west region config

Mark-H committed:
Remove the max width from the tree sidebar, so it can be resized to any width [#13637]

Merge remote-tracking branch 'upstream/pr/13756' into 2.6.x

* upstream/pr/13756:
  removed the  maxSize property from the west region config
  Addressed  issue #13309 #modxbughunt Removed the maxSize restriction on the west region allowing the splitbar to be moved further than 800px

Commit:
Static file mediasource fix

Commit:
Test template deleted

sdrenth committed:
Fixing incorrect language bug on complete step

Mark-H committed:
Fix selecting the right media source when editing a static element [#13750]

Merge remote-tracking branch 'upstream/pr/13762' into 2.6.x

* upstream/pr/13762:
  Test template deleted
  Static file mediasource fix

sdrenth committed:
Fix cookiepath

Mark-H committed:
Fix setup language being reset to English in the last step [#13611]

Merge remote-tracking branch 'upstream/pr/13755' into 2.6.x

* upstream/pr/13755:
  Fix cookiepath
  Fixing incorrect language bug on complete step
  Fixing complete step during setup would always be in EN language

bezumkin committed:
Fix silent fail on login

Return error if user has no permissions for using manager

jdaehne committed:
resolve #13740

bezumkin committed:
Prevent reload of login page

If you can't login to manager you will see an error, but the next attempt will just reload window. It is because user is actually logged it but can't see backend, so manager controller will check it and logout him on the next request. That is why we have a page reload instead of new attempt to login.

This change will fix it.

Bruno17 committed:
fix bug 13748

AmaZili committed:
Change to  bug in the parser sending to the log file unattended error message when using a placeholder containing conditions having as a result a comment (- sign)
with '#' will be ignored, and an empty message aborts the commit.

Mark-H committed:
Merge branch '2.6.x' into 2.x

Mark-H committed:
Prevent notices for undefined smarty placeholders (in TVs) [#13748]

Merge remote-tracking branch 'upstream/pr/13769' into 2.x

* upstream/pr/13769:
  fix bug 13748

Mark-H committed:
Fix incorrect view url after changing the resource url [#13740]

Merge remote-tracking branch 'upstream/pr/13768' into 2.6.x

* upstream/pr/13768:
  resolve #13740

Mark-H committed:
Fix silent fail on login without manager access [#12706]

Merge remote-tracking branch 'upstream/pr/13767' into 2.6.x

* upstream/pr/13767:
  Prevent reload of login page
  Fix silent fail on login

Mark-H committed:
Adjust the form customization sets grid to show the lexicon entry for the action

Mark-H committed:
Add wildcard (e.g. resource/*) support to form customization, allowing rules to apply to both create and update actions

With this change, FC no longer uses the for_parent value, which did not appear to do much other than being set to 1 for resource/create actions. Now, only the action is used.

The FC rules will now look for `resource/*` as well as direct matches (e.g. `resource/create` or `resource/update`). This could theoretically also be used for other actions (user? context?).

Bruno17 committed:
Prevent adding ./ to filepath when in root of a mediasource [12625]

chrisdempsey committed:
Update default.inc.php

Fix typo in $_lang['tv_elements_desc']

bezumkin committed:
Ability to create MODX project with composer

bezumkin committed:
Removed unneeded dependencies in 2.x branch

opengeek committed:
Add ability to create MODX project with composer

Merge remote-tracking branch 'origin/pr/13784' into 2.x

* origin/pr/13784:
  Removed unneeded dependencies in 2.x branch
  Ability to create MODX project with composer

rthrash committed:
Delete a few no-longer-needed images

Resolves #13788

Mark-H committed:
Remove ununused images [#13788]

Merge remote-tracking branch 'upstream/pr/13789' into 2.x

* upstream/pr/13789:
  Delete a few no-longer-needed images

Mark-H committed:
Fix incorrect hex colors mentioned in the tv input options description

Merge remote-tracking branch 'upstream/pr/13779' into 2.x

* upstream/pr/13779:
  Update default.inc.php

bezumkin committed:
CLI install script to use after composer create-project

kudashevs committed:
Make less strict https status check

BobRay committed:
Fix Bug in MessageCreate processor

Fixes bug with Message Create processor using the wrong recipient ID when processing usergroup messages.

Commit:
We don't want to replace the existing headers when preparing the headers in modRest.

Commit:
We wan't to get the "verbose" option value when we set the default request options in modRest.

matdave committed:
Update sibling template to utilize highest child ID as well as default to parent if no other sibling found.

bezumkin committed:
- Added support of lexicons
- Improved questions
- More security on using shell_exec commands
- The installation process now run by including install.php, without shell_exec.
- Replaced system calls to built-in functions

opengeek committed:
Restore behavior in Property Set UI that only saves properties modified from the default for an element

Revert "Bugfix for #12580"

This reverts commit bd61e77bfba648563f7bf0faeaab6efb2cc599dc introduced via PR #12614.

Commit:
Surrounded the body of the if statement with curly brackets.

bezumkin committed:
Change type of modResource.description column

BobRay committed:
Make manager search search by id for all objects

Error messages identify snippets and plugins by ID, not by name, and on a large site they can be very difficult to find. Occasionally, there is a need to search  for other objects by ID as well and the cost of adding that search criterion is small.
#13803

bezumkin committed:
Remove maximum length of modResource.description

bezumkin committed:
Enabled modx_browser_tree_hide_files system setting by default
Enabled caching of images from phpthumb connector by browser
Lazy load of images in media browser

bezumkin committed:
No need to check src if we delete item from array

bezumkin committed:
Added timestamp to thumbnails to break cache after change.

Commit:
Enable ACL „events“ for Access Policy „Administrator“ at install

Mark-H committed:
Fix too much data being sent along to the package install request

bezumkin committed:
Fix for #13609

Mark-H committed:
Fix incorrect media source being used on image TVs when creating new resource in different context [#13609]

Merge remote-tracking branch 'upstream/pr/13819' into 2.6.x

* upstream/pr/13819:
  Fix for #13609

opengeek committed:
Fix unit tests for use with PHPUnit 4.3.x

opengeek committed:
Replace all deprecated while/each() loops with foreach() loops

each() is deprecated in PHP 7.2 and this will solve the deprecated warnings

opengeek committed:
Update version for 2.6.0-pl release

opengeek committed:
Update version for 2.7.0-dev

opengeek committed:
Replace all deprecated while/each() loops with foreach() loops for PHP 7.2 compat

Merge branch 'each-deprecated' into 2.x

* each-deprecated:
  Replace all deprecated while/each() loops with foreach() loops
  Fix unit tests for use with PHPUnit 4.3.x

opengeek committed:
Merge remote-tracking branch 'xpdo/2.x' into 2.x

* xpdo/2.x:
  Replace all deprecated while/each() loops with foreach() loops
  Update version for 2.7.0-dev
  Fix unit tests for use with PHPUnit 4.3.x
  Update version for 2.6.0-pl release

opengeek committed:
Merge branch '2.6.x' into 2.x

* 2.6.x:
  Fix for #13609
  Prevent reload of login page
  resolve #13740
  Fix silent fail on login

opengeek committed:
Missed one instance of each() in xPDO class + some in unit tests

NOTE: also fixes unit test for embedded escape chars on sqlite/sqlsrv

opengeek committed:
Merge remote-tracking branch 'xpdo/2.x' into 2.x

* xpdo/2.x:
  Missed one instance of each() in xPDO class + some in unit tests

opengeek committed:
Update xPDO to address deprecated usage of each() for PHP 7.2

Merge remote-tracking branch 'xpdo/2.x' into each-deprecated

* xpdo/2.x:
  Missed one instance of each() in xPDO class + some in unit tests
  Replace all deprecated while/each() loops with foreach() loops
  Update version for 2.7.0-dev
  Fix unit tests for use with PHPUnit 4.3.x
  Update version for 2.6.0-pl release
  Generate engine into the map files for MySQL
  Make InnoDB the default engine for MySQL
  Add missing return statements to built-in validation classes

tonyklapatch committed:
Update old links

opengeek committed:
Replace all deprecated while/each() loops with foreach() loops for PHP 7.2 compat

opengeek committed:
Update Snoopy class in magpieRSS to replace deprecated each() calls

opengeek committed:
Update Smarty to 3.1.31

Addresses issue with deprecated warnings in PHP 7.2 from each() usage

opengeek committed:
Add events permission to Administrator policy on new installs

Merge remote-tracking branch 'origin/pr/13830' into 2.6.x

* origin/pr/13830:
  Enable ACL „events“ for Access Policy „Administrator“ at install

opengeek committed:
Remove erroneous use statements from xPDOQuery_sqlsrv class

opengeek committed:
Make json and array phptypes quotable in xPDOQuery

Changes in PDO in PHP 7.2+ seem to have made it impossible to write queries against these fields using xPDOQuery.

Resolves modxcms/xpdo#135

opengeek committed:
Merge remote-tracking branch 'xpdo/2.x' into 2.x

* xpdo/2.x:
  Make json and array phptypes quotable in xPDOQuery
  Remove erroneous use statements from xPDOQuery_sqlsrv class

JoshuaLuckers committed:
Prevent the Sint Maarten country_lang entry from overriding Seychelles.

opengeek committed:
Stop sending too much data on package install request

Merge remote-tracking branch 'origin/pr/13813' into 2.6.x

* origin/pr/13813:
  Fix too much data being sent along to the package install request

opengeek committed:
Fix sending messages to wrong recipients in message/create processor

Merge branch 'pr13796' into 2.6.x

* pr13796:
  Fix Bug in MessageCreate processor

opengeek committed:
Prevent http headers from being overwritten in modRest

Merge remote-tracking branch 'origin/pr/13797' into 2.6.x

* origin/pr/13797:
  Surrounded the body of the if statement with curly brackets.
  We don't want to replace the existing headers when preparing the headers in modRest.

opengeek committed:
Fix typo preventing verbose CURL option from being set in modRest

Merge remote-tracking branch 'origin/pr/13798' into 2.6.x

* origin/pr/13798:
  We wan't to get the "verbose" option value when we set the default request options in modRest.

opengeek committed:
Prevent logging errors for comments or empty tags

Merge remote-tracking branch 'origin/pr/13771' into 2.6.x

* origin/pr/13771:
  Change to  bug in the parser sending to the log file unattended error message when using a placeholder containing conditions having as a result a comment (- sign) with '#' will be ignored, and an empty message aborts the commit.

bezumkin committed:
Fix for #13761

opengeek committed:
Fix error with getonline processor on systems with only_full_group_by sql_mode

Merge remote-tracking branch 'origin/pr/13835' into 2.6.x

* origin/pr/13835:
  Fix for #13761

opengeek committed:
Update lexicons

opengeek committed:
Prevent adding ./ to filepath when in root of a mediasource

Merge remote-tracking branch 'origin/pr/13778' into 2.6.x

* origin/pr/13778:
  Prevent adding ./ to filepath when in root of a mediasource [12625]

opengeek committed:
Merge branch '2.6.x' into 2.x

* 2.6.x:
  Update lexicons
  Fix for #13761
  Fix Bug in MessageCreate processor
  Enable ACL „events“ for Access Policy „Administrator“ at install
  Fix too much data being sent along to the package install request
  Surrounded the body of the if statement with curly brackets.
  We wan't to get the "verbose" option value when we set the default request options in modRest.
  We don't want to replace the existing headers when preparing the headers in modRest.
  Prevent adding ./ to filepath when in root of a mediasource [12625]
  Change to  bug in the parser sending to the log file unattended error message when using a placeholder containing conditions having as a result a comment (- sign) with '#' will be ignored, and an empty message aborts the commit.

Commit:
Replace usages of each() to avoid deprecated warnings in PHP 7.2 #13829

* upstream/pr/13829:
  Update Smarty to 3.1.31
  Update Snoopy class in magpieRSS to replace deprecated each() calls
  Replace all deprecated while/each() loops with foreach() loops for PHP 7.2 compat
  Missed one instance of each() in xPDO class + some in unit tests
  Replace all deprecated while/each() loops with foreach() loops
  Update version for 2.7.0-dev
  Fix unit tests for use with PHPUnit 4.3.x
  Update version for 2.6.0-pl release
  Generate engine into the map files for MySQL
  Make InnoDB the default engine for MySQL
  Add missing return statements to built-in validation classes

Commit:
Use the correct ISO 3166 code for Sint Maarten

Commit:
Revert "Use the correct ISO 3166 code for Sint Maarten"

This reverts commit 120b111a1b6f98f61ecb45e257fa3db45cc1d49c.

Commit:
Use the correct ISO 3166 code for Sint Maarten [#13836]

* upstream/pr/13836:
  Prevent the Sint Maarten country_lang entry from overriding Seychelles.

bezumkin committed:
Remove unneeded backtick

Commit:
Change modResource.description column to text [#13802]

* upstream/pr/13802:
  Remove unneeded backtick
  Remove maximum length of modResource.description
  Change type of modResource.description column

Commit:
Remove obsole quotes around table name in messages

opengeek committed:
Cleaning up merge messes

opengeek committed:
Display context name and key in Context dropdown

This is to make it clearer when working with areas where the context key is being displayed in a grid, as in when editing context ACLs, and to provide a fallback for contexts where no name is provided.

Addresses #13739

opengeek committed:
Merge branch '2.6.x' into 2.x

* 2.6.x:
  Cleaning up merge messes
  Update Smarty to 3.1.31
  Update Snoopy class in magpieRSS to replace deprecated each() calls
  Replace all deprecated while/each() loops with foreach() loops for PHP 7.2 compat
  Missed one instance of each() in xPDO class + some in unit tests
  Replace all deprecated while/each() loops with foreach() loops
  Update version for 2.7.0-dev
  Fix unit tests for use with PHPUnit 4.3.x
  Update version for 2.6.0-pl release

opengeek committed:
Only save properties modified from the default for an element in Property Sets

Merge remote-tracking branch 'origin/pr/13799' into 2.6.x

* origin/pr/13799:
  Restore behavior in Property Set UI that only saves properties modified from the default for an element

opengeek committed:
Display context name and key in Context dropdown

Merge remote-tracking branch 'origin/pr/13839' into 2.6.x

* origin/pr/13839:
  Display context name and key in Context dropdown

opengeek committed:
Update version for 2.6.2 release

opengeek committed:
Update xPDO for a couple of xPDOQuery bugs

Merge remote-tracking branch 'xpdo/2.x' into 2.6.x

* xpdo/2.x:
  Make json and array phptypes quotable in xPDOQuery
  Remove erroneous use statements from xPDOQuery_sqlsrv class

opengeek committed:
Merge branch '2.6.x' into 2.x

* 2.6.x:
  Update version for 2.6.2 release
  Display context name and key in Context dropdown
  Restore behavior in Property Set UI that only saves properties modified from the default for an element
  Make json and array phptypes quotable in xPDOQuery
  Remove erroneous use statements from xPDOQuery_sqlsrv class

opengeek committed:
Update old links in lexicons for the about view

Merge remote-tracking branch 'origin/pr/13828' into 2.x

* origin/pr/13828:
  Update old links

opengeek committed:
Remove unnecessary backticks around table name in upgrades lexicon

Merge remote-tracking branch 'origin/pr/13838' into 2.x

* origin/pr/13838:
  Remove obsole quotes around table name in messages

opengeek committed:
Media browser optimizations

Merge remote-tracking branch 'origin/pr/13805' into 2.x

* origin/pr/13805:
  Added timestamp to thumbnails to break cache after change.
  No need to check src if we delete item from array
  Enabled modx_browser_tree_hide_files system setting by default Enabled caching of images from phpthumb connector by browser Lazy load of images in media browser

opengeek committed:
Update upgrade script for modx_browser_tree_hide_files setting

opengeek committed:
Add automatic_template_assignment feature

Merge remote-tracking branch 'origin/pr/13700' into 2.x

* origin/pr/13700:
  Update sibling template to utilize highest child ID as well as default to parent if no other sibling found.
  Changed from boolean to turn off parent inheritance to work more like Evo with the additional option of inheriting sibling templates
  Adding system setting "inherit_parent_template" to fix #5980

opengeek committed:
Add ability to search by id on all objects in manager search

Merge remote-tracking branch 'origin/pr/13804' into 2.x

* origin/pr/13804:
  Make manager search search by id for all objects

opengeek committed:
Make HTTPS server check accept any non-empty value

Merge remote-tracking branch 'origin/pr/13794' into 2.x

* origin/pr/13794:
  Make less strict https status check

opengeek committed:
Add wildcard support to form customization actions

Merge remote-tracking branch 'origin/pr/13775' into 2.x

* origin/pr/13775:
  Add wildcard (e.g. resource/*) support to form customization, allowing rules to apply to both create and update actions
  Adjust the form customization sets grid to show the lexicon entry for the action

opengeek committed:
Allow extension packages to have an empty table_prefix

Merge remote-tracking branch 'origin/pr/13716' into 2.x

* origin/pr/13716:
  Update modx.class.php

opengeek committed:
Add CODEOWNERS file

opengeek committed:
Add CODEOWNERS file

Mark-H committed:
Fix installation of transport packages with setup options

In #13813 I removed the va variable from being sent to the processor as that contained an extjs config object which was tripping over mod_security rules. But I failed to notice that setup options from a package were also being sent in this variable, so in 2.6.2 setup options were no longer being sent to the server, breaking installation (or preventing easy set-up) of some packages.

To remove the ambiguity of the va variable I've updated the setup options install method to send the same variables to install, with the options as a new third parameter instead. That way it can be treated properly. I've also added extensive comments for the next poor sap to explore into the depths of the package installation process.

Commit:
Fix installation of transport packages with setup options #13861

* upstream/pr/13861:
  Fix installation of transport packages with setup options

opengeek committed:
Lexicon updates

opengeek committed:
MODX Revolution 2.6.3

opengeek committed:
Merge branch '2.6.x' into 2.x

* 2.6.x:
  MODX Revolution 2.6.3
  Add CODEOWNERS file
  Lexicon updates
  Fix installation of transport packages with setup options

opengeek committed:
Add CLI install script for use with composer create-project

Merge remote-tracking branch 'origin/pr/13790' into 2.x

* origin/pr/13790:
  - Added support of lexicons - Improved questions - More security on using shell_exec commands - The installation process now run by including install.php, without shell_exec. - Replaced system calls to built-in functions
  CLI install script to use after composer create-project

gpsietzema commented:
@davidpede MODX 2.7 is coming soon, do you see any chance to change this PR on short term?

davidpede commented:
Can look at this next week if soon enough? @gpsietzema

gpsietzema commented:
That would work if it is early next week @davidpede - Jason is planning on merging everything in this and next week. Believe me, it's a sh*tload of work ;-)

davidpede committed:
Merge remote-tracking branch 'upstream/2.x' into tv-tpl-sort-fix

davidpede commented:
@gpsietzema @Jako I completely screwed this branch so have re-opened here #13893
